main.go

```diff
@@ -114,7 +114,7 @@ func main() {
 	case "aws":
 		p, err = provider.NewAWSProvider(cfg.DomainFilter, cfg.DryRun)
 	case "inmemory":
-		p, err = provider.NewInMemoryProviderWithDomainAndLogging("example.com"), nil
+		p, err = provider.NewInMemoryProvider(provider.InMemoryWithDomain(cfg.DomainFilter), provider.InMemoryWithLogging()), nil
```

Comment:
In this case it is probably not very clear because: 

1. `OnApplyChanges` is public, when it really should not be - or it should not be used for logging. However it is used in other packages, so it will require move refactoring, which I didn't want to do in this PR. 
2. `domain` should be set via passed parameter. But for the InMemoryProvider `domain` field` has very little value, therefore IMO it should not be a required parameter for the default initialiser (`NewInMemoryProvider`)